### ISSUE TITLE
wpt-decision-task can fail due to git race condition

### ISSUE THREAD

**Issue by foolip** [web-platform-tests/wpt#23312]
In https://github.com/web-platform-tests/wpt/pull/23283, [wpt-decision-task](https://community-tc.services.mozilla.com/tasks/DsABju6HStOsvffovetJig) failed like this:
```
INFO:root:Looking for changes in range 56c273203cb250902022702481d56539963ea2ff..6303b9e57674d4d60b182d81779bf4b4257c4e59
fatal: Invalid revision range 56c273203cb250902022702481d56539963ea2ff..6303b9e57674d4d60b182d81779bf4b4257c4e59
```

Right now, fetching the repo both commit 56c273203cb250902022702481d56539963ea2ff and 6303b9e57674d4d60b182d81779bf4b4257c4e59 can be found.

The Taskcluster run was triggered by [a force push](https://github.com/web-platform-tests/wpt/pull/23283#event-3282177246) and the problem is most likely that fetching refs/pull/23283/merge fetched a merge of the old branch. However, this can't be determined by looking at the logs.

In other words, there's like a race condition between the webhooks triggering Taskcluster and whatever GitHub-internal bits update the refs/pull/23283/merge ref.

Retrying the fetch until the expected references can be found would probably work.

@LukeZielinski do we have monitoring in place to know how often wpt-decision-task fails?

**Comment by stephenmcgruer** [comment]
cc @jgraham who has previously worked on tricky issues with wpt-decision-task and missing revisions/etc

**Comment by stephenmcgruer** [comment]
Tentatively marking roadmap, but this has blocked a half-dozen or so Chromium exports over the last week so maybe it should be 'urgent'.. Not sure if something has changed to make this more common or if its just bad luck.

**Comment by foolip** [comment]
This happened in https://github.com/web-platform-tests/wpt/pull/21553 too.

**Comment by foolip** [comment]
@stephenmcgruer this came up in the rotation tool. Have you continued to see this happen?

**Comment by stephenmcgruer** [comment]
I have not seen it recently. But I suspect you are still right and it's a race condition between Taskcluster and GitHub internals. My suggestion would be the same as yours was in April - retrying the `get_paths` call in `get_run_jobs` (in tools/ci/tc/decision.py) a few times with a small back-off.

@jgraham any opinions on this?

**Comment by stephenmcgruer** [comment]
Ping @jgraham for their thoughts, but also marking `backlog` because this doesn't seem to have happened *too* much.

**Comment by jgraham** [comment]
Feels like we should look at `get_fetch_rev` as the place to ensure the revs we get are actually sensible. And also that we should probably run that command once rather than once per scheduled task. If we ran that as soon as we've parsed the event, and had some backoff and retry solution for stale merge refs, I think that would fix things.
